### ISSUE TITLE
Request for CI to produce meaningful email notifications

### ISSUE THREAD

**Issue by mcfadden8** [LLNL/Umpire#381]
For Umpire, there are only a few reasons that CI can fail against any given project:

1. CI Success
2. Build failure (compile/link errors) - most likely
3. Test Failures
4. Git failure
5. CI Framework failure - least likely

In generatl, each email summary line should contain: <Status> <Build #> <Project> <Branch>

Status should be one of "CI Success", "Build Failure", "Test Failure", "Git Failure", "CI Framework Failure".

The content of the mail should begin with links to the applicable assets.  This should be followed with a snippet of the error logs showing what the specific error(s) is/are.

**Comment by adrienbernede** [comment]
Thanks for creating the issue.

I found a gitlab issue that tells me this is not yet possible directly with Gitlab at the moment:
https://gitlab.com/gitlab-org/gitlab/-/issues/24841

Also, a quick test this afternoon showed that even when "Watching" a project, only the user triggering the pipeline receives a notification, **for a pipeline trigger from push**. However, every pipeline is considered triggered from push when mirroring (by opposition to triggered for a Merge Request, typically).

In short, I think we are on our own to manage this. We could definitely hack something to receive notification and reporting from each pipeline. I’ll think about this, although I already have a lot of my plate at the moment.

**Comment by mcfadden8** [comment]
Not worried about this.

**Comment by adrienbernede** [comment]
I’ll be back soon, sorting issues that matters to you guys will help for sure.
